release-20.1: sql: fix TRUNCATE for temporary tables

otan opened the pull request:
Backport 1/1 commits from #47482.

/cc @cockroachdb/release

Not planning on pushing this until the clearance has been given.

---

Resolves https://github.com/cockroachdb/cockroach/issues/47030

Release note (bug fix): TRUNCATE can now run on temporary tables,
fixing a bug in 20.1.0 where temporary tables could not be truncated,
resulting in an error `unexpected value: nil`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48078)
<!-- Reviewable:end -->

otan committed:
sql: fix TRUNCATE for temporary tables

Release note (bug fix): TRUNCATE can now run on temporary tables,
fixing a bug in 20.1.0 where temporary tables could not be truncated,
resulting in an error `unexpected value: nil`.
